src/legacy/core_plugins/telemetry/server/get_telemetry_opt_in.ts

```diff
@@ -43,4 +55,60 @@ export async function getTelemetryOptIn(request: any) {
 
     throw error;
   }
+
+  const { attributes } = savedObject;
+
+  // return `enabled` attribute, unless it's false ...
+  if (attributes.enabled !== false) return attributes.enabled;
+
+  // Additional check if they've already opted out (enabled: false):
+  // - if the Kibana version has changed by at least a minor version,
+  //   set them back to the null state to get prompted again.
+
+  const lastKibanaVersion = attributes.lastVersionChecked;
+
+  // if version hasn't changed, just return enabled value
+  if (lastKibanaVersion === currentKibanaVersion) return attributes.enabled;
+
+  const lastSemver = parseSemver(lastKibanaVersion);
+  const currentSemver = parseSemver(currentKibanaVersion);
+
+  // if both versions aren't valid, just return the current enabled value
+  if (lastSemver == null && currentSemver == null) {
+    return attributes.enabled;
+  }
```

Comment:
nit, but feels like these two can actually be a single `return or continue` clause, rather than two returns.
I realise you're probably trying to avoid running `parseSemver` unnecessarily, but there are multiple returns from this function, which makes it harder to follow, so removing even one return could help with maintainability.

Reply:
there are TONS of returns in this function :-)

are you talking about these two?

- `if (lastSemver == null && currentSemver == null)`
- `if (lastSemver == null || currentSemver == null)`

Somewhat arbitrary decision by me to treat these differently.  Probably removing the first check is fine, making it if either the last or current version wasn't valid, re-prompt.

Reply:
ended up removing the 1st test - now just tests `||` and returns null in that case.